### ISSUE TITLE
Support 40 MHz bandwidth on Wi-Fi

### ISSUE THREAD

**Issue by aravindvnair99** [PixelExperience/android-issues#1812]
Many devices support 40 MHz. Would be nice to see this native within Pixel Experience. For example, I have a Xiaomi POCO F1 which when I enable 40 MHz gets higher bandwidth hence taking 2.4 GHz Wi-Fi to 300 Mbps than the default 150 Mbps.

Refer to this for more details: https://github.com/Magisk-Modules-Repo/wifi-bonding

**Comment by pixelexperiencebot** [comment]
Validation failed, this issue will be closed for not complying with the established standards.

Errors:
- Missing parameter: device
- Missing parameter: version
- Invalid issue template

**Comment by aravindvnair99** [comment]
@Hlcpereira and @jhenrique09 Where do I put up feature requests/suggestions?

**Comment by jjpprrrr** [comment]
2.4GHz WiFi channel bonding being disabled is an intentional engineering choice, not an issue of device not supporting it. Here is my reasoning:
- In most countries or regions, 2.4G WiFi can only legally use a frequency range of 72MHz - from 2.401GHz to 2.473GHz. For IEEE802.11n (WiFi 5, which is 2.4GHz mostly used for), every channel is 20MHz, and there are 11 channels in total. Therefore, only channel No.1, 6, and 11 are completely not overlapped. The rest of the channels have some overlapping frequency with the nearby channels, causing ACI (Adjacent Channel Interference)
- ACI is bad. Interference can cause significant packet loss and bandwidth reduction for you WiFi.
- What is channel bonding? It combines two 20MHz channels together to increase the theoretical bandwidth. This is the so-called "40MHz" you mentioned in your title.
- Why is channel bonding for 2.4G WiFi turned off? 40Mhz out of 72MHz total range means it's basically impossible for the 40MHz to not have interference with any adjacent channels, thus causing bad WiFi experience in real life. It can also make the area more congested and affects other devices' connections.
- The 2.4GHz range is already super congested in most cities. We do not want to make it worse. The 2.4G WiFi is good at covering a larger area, not providing a higher bandwidth. If you need bandwidth, use 5G WiFi.
- Many cellphone OEMs disable 2.4G WiFi channel bonding intentionally, including Google, Xiaomi, Oppo/OnePlus, Nokia, etc. They are probably doing it for a reason, right?
- If you are living in the middle of nowhere, and you only have one WiFi source in your local area, you can always use that module to re-enable channel bonding for 2.4G WiFi.
- The channel bonding is always enabled for 5G WiFi, and again, if you really need that bandwidth, you should probably use 5G WiFi.

**Comment by jjpprrrr** [comment]
> @Hlcpereira and @jhenrique09 Where do I put up feature requests/suggestions?

Please follow the issue template and learn to submit an issue properly next time. Tagging people randomly is not going to help you.

**Comment by aravindvnair99** [comment]
Hey @jjpprrrr!

Thanks a lot for the quick response and the detailed explanation. I agree with every point you stated, but I feel it would be good to have it as a toggle with the default being set to `Off` as depending on the locality, it can be an advantage or disadvantage.

> If you are living in the middle of nowhere, and you only have one WiFi source in your local area, you can always use that module to re-enable channel bonding for 2.4G WiFi.

There are many Wi-Fi networks in my place (almost 20 - 30 networks), but I have noticed 40 MHz has a significant improvement to my network activities especially due to all houses being well spaced. But yes, I agree it varies from location to location.

I had raised this feature request as I had seen every modern-day 802.11n and above laptop support 40 MHz for 2.4 GHz Wi-Fi, but not phones although this feature was introduced with 802.11n. So felt, it would be nice to have a toggle than manually edit configuration files when the need arises. I do have 5 GHz as well, and yes channel bonding works perfectly on that. But sometimes when further away, switching to 2.4 GHz away is better and there the 40 MHz truly outshines. Hence, in my opinion, having a toggle is better than having no toggle.

> Please follow the issue template and learn to submit an issue properly next time. Tagging people randomly is not going to help you.

I understand. Unfortunately, I only see a bug report template and not a feature request template. Hence, I raised this ticket using the blank template here: https://github.com/PixelExperience/android-issues/issues/new/choose. That's why I asked where I could request/suggest features that could be considered to be added in.

Thanks again for the quick response!
